pkg/storegateway/indexheader/index/postings.go

```diff
@@ -403,13 +441,29 @@ func (t *PostingOffsetTableV2) LabelValues(name string, filter func(string) bool
 		} else {
 			d.Skip(skip)
 		}
-		s := yoloString(d.UnsafeUvarintBytes()) // Label value.
-		if filter == nil || filter(s) {
-			// Clone the yolo string since its bytes will be invalidated as soon as
-			// any other reads against the decoding buffer are performed.
-			values = append(values, strings.Clone(s))
+
+		value := yoloString(d.UnsafeUvarintBytes())
```

Comment:
[nit] I suggest to rename `value` to `unsafeValue`. It makes more clear when we use yolo strings.